api/resolvers/notifications.js

```diff
@@ -330,6 +332,22 @@ export default {
         LIMIT ${LIMIT})`
       )
 
+      queries.push(
+        `(SELECT "Invoice".id::text, "Invoice"."updated_at" AS "sortTime", NULL as "earnedSats", 'Invoicification' AS type
+        FROM "Invoice"
+        WHERE "Invoice"."userId" = $1
+        AND "Invoice"."updated_at" < $2
+        AND "Invoice"."actionState" = 'FAILED'
+        AND (
+          "Invoice"."actionType" = 'ITEM_CREATE' OR
+          "Invoice"."actionType" = 'ZAP' OR
+          "Invoice"."actionType" = 'DOWN_ZAP' OR
+          "Invoice"."actionType" = 'POLL_VOTE'
+        )
+        ORDER BY "sortTime" DESC
+        LIMIT ${LIMIT})`
+      )
+
```

Comment:
Without someone explicitly hitting a `cancel` button it feels weird to assume they meant to cancel when they closed a modal. The binary success/failure is a lot easier to think through too. 

The approach I took was "they tried to pay and the flow didn't complete" = failed.

----------

One improvement on retries that I'd like to make is doing "implicit retries" where if you repeat a failed action like you do in your video (without hitting the `retry` button), the backend recognizes it's repeating something that failed and does a retry rather than creating a new action.